tests/t0_http.py

```diff
@@ -701,12 +732,33 @@ def do_GET(self):
     conn.send_request('GET', path)
     resp = conn.read_response()
     assert resp.status == 200
-    with pytest.raises(UnsupportedResponse):
+    with pytest.raises(InvalidResponse):
         conn.readall()
     assert conn.read_raw(512) == b'body data'
     assert conn.read_raw(512) == b''
 
 
+def test_missing_content_length(conn, monkeypatch):
```

Comment:
Could you add a comment explaining how this differs from the newly added test_conn_close_2 above?

Reply:
I have added the comment and changed the test to also test that the injected/corrected `Connection: close` header ist there: https://github.com/s3ql/s3ql/compare/dd989152843e6efa8e900e29dfcccdc81c512759..7db30c64261a9674b0c3b091176ebd2630e038e4